perma_web/api2/utils.py

```diff
@@ -0,0 +1,134 @@
+import imghdr
+from collections import OrderedDict
+from functools import wraps
+
+from django.http import Http404
+from rest_framework import serializers
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.pagination import LimitOffsetPagination
+from rest_framework.response import Response
+from rest_framework.settings import api_settings
+
+from perma.models import Folder
+
+
+class TastypiePagination(LimitOffsetPagination):
+    """
+        Modify DRF's LimitOffsetPagination to return results in the same format as paginated results returned by Tastypie.
+    """
+    def get_paginated_response(self, data):
+        return Response(OrderedDict([
+            ('meta', OrderedDict([
+                ('limit', self.limit),
+                ('next', self.get_next_link()),
+                ('offset', self.offset),
+                ('previous', self.get_previous_link()),
+                ('total_count', self.count)
+            ])),
+            ('objects', data)
+        ]))
+
+
+def raise_validation_error(message):
+    raise serializers.ValidationError({
+        api_settings.NON_FIELD_ERRORS_KEY: [message]
+    })
+
+
+def log_api_call(func):
+    """
+        Handy function to wrap around view methods to log input and output for debugging.
+    """
+    @wraps(func)
+    def func_wrapper(self, request, *args, **kwargs):
+        print func.__name__, "called with", request, request.data, args, kwargs
+        try:
+            result = func(self, request, *args, **kwargs)
+        except Exception as e:
+            print "returning exception:", e
+            raise
+        print "returning to user", request.user, result.status_code, result.data
+        return result
+    return func_wrapper
+
+
+parent_classes = {
+    'folders': Folder,
+}
+def load_parent(func):
+    """
+        Decorator to set request.parent for nested views. For example, if we have
+
+            /folders/1/archives/2
+
+        And
+
+            class LinkDetailView():
+
+                @load_parent
+                def get(request, pk):
+                    ...
+
+        This decorator will make sure that request.parent is set to Folder(pk=2),
```

Comment:
Typo, I think. Folder(pk=1). Also, triviality: maybe use /folders/1/folders/2 as an example (since archive IDs are GUIDS rather than ints, and since you can access /folders/int/folders/int with a  GET request)